### ISSUE TITLE
Tree view and tree model APIs should be simmilar

### ISSUE THREAD

**Issue by pjasiun** [ckeditor/ckeditor5-engine#165]
Both data structures use the thee structures which allows attributes and child nodes manipulating. We should ensure that if both have `insertChildren` method, then both of these methods have the same order of parameters.

AFAIR, beside https://github.com/ckeditor/ckeditor5-core/issues/76, we need to introduced `element.appendChildren` to the tree model element.

**Comment by scofalik** [comment]
Right now, view and model API differs a bit. I though this will not be painful, but writing converters showed me that it actually is.

Spotted differences:
* `ViewElement` should be able to get a string as child(ren) instead of `ViewText`, like `ModelElement`
* `getAttributeKeys` + `getAttribute` in view vs `getAttributes` in model
* #351 - `ModelElement.getChildren()`

In docs:
* We have something called `engine.treeModel.Item`. When documenting I noticed that something similar could be useful for view, but I am not sure whether it would be applicable in all places in view so this is to-be-researched.

If you spotted or remember any other differences, let me know here.

**Comment by scofalik** [comment]
In #352 we implemented `ModelElement.getText()`. But we are missing `ViewElement.getText()`. Also I don't like that we have `.text` in `ModelTextProxy`, `.character` in `CharacterProxy` and `.getText()` in `ModelElement`. We also have to think to make it right in view if we are going to implement `ViewElement.getText()` in view.

**Comment by pjasiun** [comment]
Related: https://github.com/ckeditor/ckeditor5-engine/issues/511 Model Selection setTo method is missing

**Comment by pjasiun** [comment]
Related: https://github.com/ckeditor/ckeditor5-engine/issues/518 Decide where we should use getters, where methods

**Comment by scofalik** [comment]
For now I decided to skip getters/setters unification across model and view. It can be done in #518. I only brought missing API parts but some of them are getters in model and methods in view and vice versa.

I also wasn't really sure what to do about `Node#getPath()` in model vs `Node#getAncestors()` in view. I ended up not adding `getAncestors` to model and `getPath` to view. I assumed they do similar thing, and since model positions bases on paths while view positions bases on element+offset it makes sense.
